failed Folder extractions are never retried

Origin user: nyaaaaaah
Origin message: [Discord Server Link](https://discord.com/channels/184509568867368963/781625367391371274/1224118969774899330)
Reporting user: twitchcaptain
Reporting message: [Discord Server Link](https://discord.com/channels/184509568867368963/781625367391371274/1224152966399791285)

```
[INFO] 2024/03/31 18:54:02 Unpackerr v0.13.1 Starting! PID: 1, UID: 1000, GID: 1000, Now: 2024-03-31 18:54:03 +0200 SAST
[INFO] 2024/03/31 18:54:02 ==> GoLift Discord: https://golift.io/discord <==
[INFO] 2024/03/31 18:54:02 ==> Startup Settings <==
[INFO] 2024/03/31 18:54:02  => Sonarr Config: 0 servers
[INFO] 2024/03/31 18:54:02  => Radarr Config: 0 servers
[INFO] 2024/03/31 18:54:02  => Lidarr Config: 0 servers
[INFO] 2024/03/31 18:54:02  => Readarr Config: 0 servers
[INFO] 2024/03/31 18:54:02  => Folder Config: 1 path: /downloads, extract to: /downloads/watchfromhere (delete after:0s, delete orig:false, log file: true, move back:false, isos:false, event buffer:20000)
[INFO] 2024/03/31 18:54:02  => Using env variables only. Config file not found.
[INFO] 2024/03/31 18:54:02  => Parallel: 1
[INFO] 2024/03/31 18:54:02  => Passwords: 0 (rar/7z)
[INFO] 2024/03/31 18:54:02  => Interval: 2m
[INFO] 2024/03/31 18:54:02  => Start Delay: 1m
[INFO] 2024/03/31 18:54:02  => Retry Delay: 5m, max: 3
[INFO] 2024/03/31 18:54:02  => Debug / Quiet: false / false
[INFO] 2024/03/31 18:54:02  => Activity / Queues: false / 1m
[INFO] 2024/03/31 18:54:02  => Directory & File Modes: 0777 & 0666
[INFO] 2024/03/31 18:54:02  => Log File: /downloads/unpackerr/unpackerr.log (10 @ 10Mb)
[INFO] 2024/03/31 18:54:02  => Webhook Configs: 0 URLs
[INFO] 2024/03/31 18:54:02  => Command Hook Configs: 0 commands
[INFO] 2024/03/31 18:54:02  => Webserver Disabled
[INFO] 2024/03/31 18:54:02 [Folder] Watching (fsnotify): /downloads
[INFO] 2024/03/31 18:54:02 [Folder] Polling @ 1s: /downloads
<snip>
[INFO] 2024/03/31 22:48:02 [Unpackerr] Totals: [0 retries] [0 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:48:23 [Folder] Tracking New Item: /downloads/The.Rookie.S06E02.1080p.WEB.H264-SuccessfulCrab (event: f CREATE)
[INFO] 2024/03/31 22:49:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:49:02 [Unpackerr] Totals: [0 retries] [0 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:49:27 [Folder] Queued: /downloads/The.Rookie.S06E02.1080p.WEB.H264-SuccessfulCrab, queue size: 1
[INFO] 2024/03/31 22:49:27 [Folder] Extraction Started: /downloads/The.Rookie.S06E02.1080p.WEB.H264-SuccessfulCrab, items in queue: 0
[INFO] 2024/03/31 22:49:27 Deleted (recursively): /downloads/watchfromhere/The.Rookie.S06E02.1080p.WEB.H264-SuccessfulCrab_unpackerred
[ERROR] 2024/03/31 22:49:27 [Folder] Extraction Failed: /downloads/The.Rookie.S06E02.1080p.WEB.H264-SuccessfulCrab: rardecode.OpenReader: rardecode: RAR signature not found
[INFO] 2024/03/31 22:49:37 [Folder] Finished, Removed History: /downloads/The.Rookie.S06E02.1080p.WEB.H264-SuccessfulCrab
[INFO] 2024/03/31 22:50:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:50:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:51:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:51:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:52:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:52:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:53:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:53:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:54:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:54:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:55:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:55:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:56:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:56:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:57:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:57:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:58:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:58:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 22:59:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 22:59:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:00:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:00:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:01:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:01:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:02:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:02:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:03:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:03:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:04:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:04:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:05:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:05:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:06:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:06:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:06:35 [Folder] Tracking New Item: /downloads/The.Rookie.S06E02.720p.HDTV.x264-SYNCOPY (event: f CREATE)
[INFO] 2024/03/31 23:07:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:07:02 [Unpackerr] Totals: [0 retries] [1 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:07:37 [Folder] Queued: /downloads/The.Rookie.S06E02.720p.HDTV.x264-SYNCOPY, queue size: 1
[INFO] 2024/03/31 23:07:37 [Folder] Extraction Started: /downloads/The.Rookie.S06E02.720p.HDTV.x264-SYNCOPY, items in queue: 0
[INFO] 2024/03/31 23:07:37 Deleted (recursively): /downloads/watchfromhere/The.Rookie.S06E02.720p.HDTV.x264-SYNCOPY_unpackerred
[ERROR] 2024/03/31 23:07:37 [Folder] Extraction Failed: /downloads/The.Rookie.S06E02.720p.HDTV.x264-SYNCOPY: rardecode.OpenReader: rardecode: RAR signature not found
[INFO] 2024/03/31 23:07:47 [Folder] Finished, Removed History: /downloads/The.Rookie.S06E02.720p.HDTV.x264-SYNCOPY
[INFO] 2024/03/31 23:08:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:08:02 [Unpackerr] Totals: [0 retries] [2 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:09:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:09:02 [Unpackerr] Totals: [0 retries] [2 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:10:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:10:02 [Unpackerr] Totals: [0 retries] [2 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:11:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:11:02 [Unpackerr] Totals: [0 retries] [2 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:12:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:12:02 [Unpackerr] Totals: [0 retries] [2 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
[INFO] 2024/03/31 23:13:02 [Unpackerr] Queue: [0 waiting] [0 queued] [0 extracting] [0 extracted] [0 imported] [0 failed] [0 deleted]
[INFO] 2024/03/31 23:13:02 [Unpackerr] Totals: [0 retries] [2 finished] [0|0 webhooks] [0|0 cmdhooks] [stacks; event:0, hook:0, del:0]
```

--- a/pkg/unpackerr/folder.go
+++ b/pkg/unpackerr/folder.go
@@ -441,6 +441,8 @@ func (u *Unpackerr) checkFolderStats() {
 			folder.step = WAITING
 			u.Printf("[Folder] Re-starting Failed Extraction: %s (%d/%d, failed %v ago)",
 				folder.cnfg.Path, folder.retr, u.MaxRetries, elapsed.Round(time.Second))
+		case EXTRACTFAILED == folder.step && folder.retr < u.MaxRetries:
+			// This empty block is to avoid deleting an item that needs more retries.
 		case folder.step > EXTRACTING && folder.cnfg.DeleteAfter.Duration <= 0:
 			// if DeleteAfter is 0 we don't delete anything. we are done.
 			u.updateQueueStatus(&newStatus{Name: name, Status: DELETED, Resp: nil}, false)

--- a/pkg/unpackerr/logs.go
+++ b/pkg/unpackerr/logs.go
@@ -236,7 +236,7 @@ func (u *Unpackerr) logStartupInfo(msg string) {
 	u.Printf(" => Parallel: %d", u.Config.Parallel)
 	u.Printf(" => Passwords: %d (rar/7z)", len(u.Config.Passwords))
 	u.Printf(" => Interval: %v", u.Config.Interval)
-	u.Printf(" => Start Delay: %v", u.Config.StartDelay)
+	u.Printf(" => Start/Delete Delay: %v/%v", u.Config.StartDelay, u.Config.DeleteDelay)
 	u.Printf(" => Retry Delay: %v, max: %d", u.Config.RetryDelay, u.Config.MaxRetries)
 	u.Printf(" => GUI / StdErr: %v / %v", ui.HasGUI(), u.ErrorStdErr)
 	u.Printf(" => Debug / Quiet: %v / %v", u.Config.Debug, u.Config.Quiet)

